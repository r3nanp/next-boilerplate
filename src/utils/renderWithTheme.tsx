import { ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'

import { theme } from 'styles/theme'

export const renderWithTheme = (children: ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
