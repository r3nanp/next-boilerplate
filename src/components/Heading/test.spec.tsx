import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/renderWithTheme'
import { Heading } from '.'

describe('<Heading />', () => {
  it('should render a white heading by default', () => {
    const { container } = renderWithTheme(<Heading>Hello world</Heading>)
    expect(screen.getByRole('heading', { name: /hello world/i })).toHaveStyle({
      color: '#FAFAFA'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
