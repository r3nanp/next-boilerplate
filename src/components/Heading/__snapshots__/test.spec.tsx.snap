// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render a white heading by default 1`] = `
.c0 {
  color: #FAFAFA;
  size: 1.6rem;
}

<h2
  class="c0"
  color="white"
>
  Hello world
</h2>
`;
